main/integration-tests/__test__/api-tests/key-pairs/activate-key-pair.test.js

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License").
+ *  You may not use this file except in compliance with the License.
+ *  A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ *  or in the "license" file accompanying this file. This file is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *  express or implied. See the License for the specific language governing
+ *  permissions and limitations under the License.
+ */
+
+const { runSetup } = require('../../../support/setup');
+const errorCode = require('../../../support/utils/error-code');
+
+describe('Get key pair', () => {
+  let setup;
+  let adminSession;
+
+  beforeAll(async () => {
+    setup = await runSetup();
+    adminSession = await setup.defaultAdminSession();
+  });
+
+  afterAll(async () => {
+    await setup.cleanup();
+  });
+
+  const userRoles = ['admin', 'researcher', 'guest', 'internal-guest'];
+  const nonAdminUserRoles = ['researcher', 'guest', 'internal-guest'];
+
+  describe('Activate key pair', () => {
+    it('should fail for anonymous user', async () => {
+      const keyPair = await adminSession.resources.keyPairs.create();
+      await adminSession.resources.keyPairs.keyPair(keyPair.id).deactivate({ rev: 0 });
+      const anonymousSession = await setup.createAnonymousSession();
+      await expect(anonymousSession.resources.keyPairs.keyPair(keyPair.id).activate({ rev: 1 })).rejects.toMatchObject({
+        code: errorCode.http.code.badImplementation,
+      });
```

Comment:
It might be a good idea to add one more expect to check that the key is still deactivated. This check will ensure that if we had a bug in the backend where we return bad implementation but still activate the key, this additional check will detect the bug.